### PR TITLE
showing the share button on plan page only if user has access

### PR DESCRIPTION
**Pull request by pflopez**

### CODE REVIEW

File: src/interface/src/app/shared/nav-bar/nav-bar.component.ts [hunk 1]
```diff
@@ -24,11 +27,19 @@ export class NavBarComponent {
   hasSharePlanFeatureFLag =
```
Comment (lastminutediorama) [inline src/interface/src/app/shared/nav-bar/nav-bar.component.ts:27]: do you mean for this to have a capital L in Flag?

File: src/interface/src/app/types/plan.types.ts [hunk 2]
```diff
@@ -14,6 +14,8 @@ export interface Plan extends BasePlan {
   area_acres: number;
   area_m2: number;
   creator: string;
+  role?: string;
+  permissions?: string[];
```
Comment (lastminutediorama) [inline src/interface/src/app/types/plan.types.ts:18]: just curious: are you just explicitly including all of these properties for readability? I know you want to tidy up this whole thing.

File: src/interface/src/app/shared/nav-bar/nav-bar.component.html [hunk 3]
```diff
@@ -89,7 +89,11 @@
       class="action-button"
       (click)="sharePlan()"
       data-id="share"
-      *ngIf="area === 'SCENARIOS' && hasSharePlanFeatureFLag">
+      *ngIf="
+        area === 'SCENARIOS' &&
+        hasSharePlanFeatureFLag &&
+        (canShowSharePlan$ | async)
```
Comment (lastminutediorama) [inline src/interface/src/app/shared/nav-bar/nav-bar.component.html:95]: nit: the name was confusing to me (and maybe it's only confusing to me :D ), but if this is just for the button, maybe we could call it `displaySharePlanBtn` or `displaySharePlanButton` or, alternatively, maybe just `canSharePlan` if we'd ever use this var for other things?

**Review by lastminutediorama — APPROVED** [review]
looks great, I only had very minor questions and a nit/question that you can completely ignore

File: src/interface/src/app/shared/nav-bar/nav-bar.component.ts (hunk 1 above)
Comment (pflopez) [reply src/interface/src/app/shared/nav-bar/nav-bar.component.ts:27]: thanks!

File: src/interface/src/app/shared/nav-bar/nav-bar.component.html (hunk 3 above)
Comment (pflopez) [reply src/interface/src/app/shared/nav-bar/nav-bar.component.html:95]: yeah good point, not a great name. Will update, thanks!

**Pull request merged**